ADR/0047-movable-build-tests.md

```diff
@@ -0,0 +1,34 @@
+# 47. Attestable Build-Time Tests in Integration Service
+
+Date: 2025-04-06
+
+## Status
+
+Proposed
+
+## Context
+
+Some of the security scans run during the build pipeline take a long time to
+run. This forces users to wait longer to be notified of the results of their
+builds.  In order to receive more immediate feedback, users have requested that
+they be able to run security scans during their integration tests.
+
+## Decision
+
+We will add support for adding attestations to images from their integration
```

Comment:
One thing to keep in mind is that we need to ensure conforma will behave as expected when there are two (different) SLSA Provenance attestations attached to the same image. Conforma ought to handle this, but previous experiments have shown that it does not handle this gracefully. I believe one of the issues was that it wanted to find every required build task in each of the attestations.

Reply:
> We also need to ensure that the integration test attestation does not "overwrite" the build pipeline attestation by using the same tag

I'm confused as to how this might happen.  Can you give me an example? Also, do you mean the image tags (e.g. `latest`, `v1`, etc) or a tag specifically related to the attestations?